src/DynamoCoreWpf/PerVertexMeshGeometryModel3D.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Linq;
+
+using HelixToolkit.Wpf.SharpDX;
+
+using SharpDX;
+using SharpDX.Direct3D11;
+
+using Buffer = SharpDX.Direct3D11.Buffer;
+
+namespace Dynamo.Wpf
+{
+    public class PerVertexMeshGeometryModel3D : MeshGeometryModel3D
+    {
+        public override void Attach(IRenderHost host)
+        {
+            // --- attach
+            base.Attach(host);
+
+            this.renderTechnique = Techniques.RenderPerVertexPhong;
+
+            if (this.Geometry == null)
+                return;
+
+            // --- get variables
+            this.vertexLayout = EffectsManager.Instance.GetLayout(this.renderTechnique);
+            this.effectTechnique = effect.GetTechniqueByName(this.renderTechnique.Name);
+
+            // --- transformations
+            this.effectTransforms = new EffectTransformVariables(this.effect);
+
+            // --- material 
+            this.AttachMaterial();
+
+            // --- scale texcoords
+            var texScale = TextureCoodScale;
+
+            // --- get geometry
+            var geometry = this.Geometry as MeshGeometry3D;
+
+            // -- set geometry if given
+            if (geometry != null)
+            {
+                /// --- init vertex buffer
+                this.vertexBuffer = Device.CreateBuffer(BindFlags.VertexBuffer, DefaultVertex.SizeInBytes, this.CreateDefaultVertexArray());
+
+                /// --- init index buffer
+                this.indexBuffer = Device.CreateBuffer(BindFlags.IndexBuffer, sizeof(int), this.Geometry.Indices.ToArray());
+            }
+            else
+            {
+                throw new Exception("Geometry must not be null");
+            }
+
+            /// --- init instances buffer            
+            this.hasInstances = (this.Instances != null) && (this.Instances.Any());
+            this.bHasInstances = this.effect.GetVariableByName("bHasInstances").AsScalar();
+            if (this.hasInstances)
+            {
+                this.instanceBuffer = Buffer.Create(this.Device, this.instanceArray, new BufferDescription(Matrix.SizeInBytes * this.instanceArray.Length, ResourceUsage.Dynamic, BindFlags.VertexBuffer, CpuAccessFlags.Write, ResourceOptionFlags.None, 0));
+            }
+
+            /// --- set rasterstate
+            this.OnRasterStateChanged(this.DepthBias);
+
+            /// --- flush
+            this.Device.ImmediateContext.Flush();
```

Comment:
I think you said this was a "cut & paste job"?  Let's remove the superfluous comments here.